openbas-api/src/test/java/io/openbas/service/ScenarioServiceTest.java

```diff
@@ -0,0 +1,127 @@
+    @Autowired
+    private ArticleRepository articleRepository;
+
+    @Autowired
+    InjectRepository injectRepository;
+
+    @Mock
+    GrantService grantService;
+    @Mock
+    VariableService variableService;
+    @Mock
+    ChallengeService challengeService;
+    @Autowired
+    private TeamService teamService;
+    @Mock
+    FileService fileService;
+    @Autowired
+    private InjectDuplicateService injectDuplicateService;
+
+
+    @InjectMocks
+    private ScenarioService scenarioService;
+
+    @BeforeEach
+    void setUp() {
+        scenarioService = new ScenarioService(scenarioRepository, teamRepository, userRepository, documentRepository,
+                scenarioTeamUserRepository, articleRepository, grantService, variableService, challengeService,
+                teamService, fileService, injectDuplicateService
+        );
+    }
+
+    @DisplayName("Should create new contextual teams during scenario duplication")
+    @Test
+    @Transactional(rollbackOn = Exception.class)
+    void createNewContextualTeamsDuringScenarioDuplication(){
+        // -- PREPARE --
+        List<Team> scenarioTeams = new ArrayList<>();;
+        Team contextualTeam = createTeam("fakeTeamName1", true);
+        scenarioTeams.add(contextualTeam);
+        Team noContextualTeam = createTeam("fakeTeamName2",false);
+        scenarioTeams.add(noContextualTeam);
+
+        Scenario scenario = createScenario(scenarioTeams);
+
+        // -- EXECUTE --
+        Scenario scenarioDuplicated = scenarioService.getDuplicateScenario(scenario.getId());
+
+        // -- ASSERT --
+        assertNotEquals(scenario.getId(), scenarioDuplicated.getId());
+        assertEquals(scenario.getFrom(), scenarioDuplicated.getFrom());
+        assertEquals(2, scenarioDuplicated.getTeams().size());
+        scenarioDuplicated.getTeams().forEach(team -> {
+            if (team.getContextual()){
+                assertNotEquals(contextualTeam.getId(), team.getId());
+                assertEquals(contextualTeam.getName(), team.getName());
+            } else {
+                assertEquals(noContextualTeam.getId(), team.getId());
+            }
+        });
+        assertEquals(1, scenarioDuplicated.getInjects().size());
+        assertEquals(2, scenario.getInjects().get(0).getTeams().size());
+        scenarioDuplicated.getInjects().get(0).getTeams().forEach(injectTeam -> {
+            if (injectTeam.getContextual()){
+                assertNotEquals(contextualTeam.getId(), injectTeam.getId());
+                assertEquals(
+                        scenarioDuplicated.getTeams().stream().filter(team -> team.getContextual().equals(true)).findFirst().orElse(new Team()).getId(),
+                        injectTeam.getId()
+                );
+            } else {
+                assertEquals(noContextualTeam.getId(), injectTeam.getId());
+            }
+        });
+    }
+
+    private Team createTeam(@NotBlank String name, Boolean isContextualTeam){
```

Comment:
You can add this in TeamFixture

Reply:
yes my bad i saw the utils/fixtures folder only this morning i will correct that